cmd/createorupdate/createorupdate.go

```diff
@@ -249,8 +321,16 @@ func main() {
 			log.Fatal(err)
 		}
 	}
+	var vnet string
+	if !isDelete && *useProd && *createPeerVnet != "" {
```

Comment:
If we are only planning to test this in production then you don't need to deploy the VNET stuff first, then get the vnet string and set it in the OSA deployment. You can follow @jim-minter's advice and add the OSA resource as part of your ARM template, then you should be able to share the VNET string in the ARM template?